### PR TITLE
RNN-T training for yesno.

### PR DESCRIPTION
**Pull request by csukuangfj**
```bash
$ ./transducer/decode.py


2021-12-06 16:50:33,363 INFO [asr_datamodule.py:218] About to get test cuts
2021-12-06 16:50:33,363 INFO [asr_datamodule.py:248] About to get test cuts
2021-12-06 16:50:33,956 INFO [decode.py:188] batch 0/?, cuts processed until now is 3
2021-12-06 16:50:36,368 INFO [decode.py:226] The transcripts are stored in transducer/exp-3/recogs-test_set.txt
2021-12-06 16:50:36,369 INFO [utils.py:387] [test_set] %WER 32.92% [79 / 240, 5 ins, 37 del, 37 sub ]
2021-12-06 16:50:36,370 INFO [decode.py:234] Wrote detailed error stats to transducer/exp-3/errs-test_set.txt
2021-12-06 16:50:36,371 INFO [decode.py:306] Done!
```

A very simple RNN-T training pipeline for the yesno dataset.  It uses greedy beam search for decoding.

Useful for the librispeech dataset.

### CODE REVIEW

**Comment by csukuangfj** [comment]
Note: It requires torchaudio >= v0.10.0 as it is using `rnnt_loss` from torchaudio.

**Comment by danpovey** [comment]
Wow, great!

File: egs/yesno/ASR/transducer/jointer.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+class Jointer(nn.Module):
```
Comment (pzelasko) [inline egs/yesno/ASR/transducer/jointer.py:22]: Is it Jointer or Joiner?

Comment (csukuangfj) [reply egs/yesno/ASR/transducer/jointer.py:22]: I find that both names are used.

## Joiner
- https://research.fb.com/wp-content/uploads/2021/05/Deep-Shallow-Fusion-for-RNN-T-Personalization.pdf
- https://www.cc.gatech.edu/classes/AY2021/cs7643_spring/assets/L24_rnnt_asr_tutorial_gt.pdf

## Jointer
- https://arxiv.org/pdf/2011.01576.pdf

---

I was using Joiner but I always type jointer so jointer is used. 

Do you think Joiner is more suitable here?

Comment (pzelasko) [reply egs/yesno/ASR/transducer/jointer.py:22]: I thought it’s a typo but now I find it’s actually a valid word 👀 just ignore my comment 😉

https://en.m.wikipedia.org/wiki/Jointer

**Comment by danpovey** [comment]
Should be joiner, not jointer.

**Comment by csukuangfj** [comment]
> Should be joiner, not jointer.

Thanks. Fixed.

---

Merging and will add a RNN-T training pipeline for LibriSpeech based on this PR.

**Pull request merged**